### ISSUE TITLE
Adding items to an array inside of Text doesn't change layout

### ISSUE THREAD

**Issue by DerpyCrabs** [vadimdemedes/ink#319]
When I add items to an array inside of Text new items don't render or overwrite text after it.
I also got incorrect wrapping of text in the similar layout (new array items wrap to new line even if where is space available inside of parent Box).

Is it a bug or I misunderstood text wrapping or box width prop?

Ink version: latest commit from master.
React version: 16.13.1.

Reproduction where 'text' is being overwritten by array items:
```javascript
const Test = () => {
  const [letters, setLetters] = React.useState(['a', 'b', 'c'])
  useInput((input, key) => {
    setLetters((letters) => [...letters, 'd'])
  })
  return (
    <Box width={30}>
      <Text>
        {letters.map((letter, i) => (
          <Text key={i}>{letter}</Text>
        ))}
        <Text>text</Text>
      </Text>
    </Box>
  )
}
```

Reproduction where Text is incorrectly wrapped even if where is space available inside of a parent:
```javascript
const Test = () => {
  const [letters, setLetters] = React.useState(['a', 'b', 'c'])
  useInput((input, key) => {
    setLetters((letters) => [...letters, 'd'])
  })
  return (
    <Box flexDirection='column' minHeight={6} width={30}>
      <Box width={30}>
        <Text>
            {letters.map((letter, i) => (
              <Text key={i}>{letter}</Text>
            ))}
        </Text>
      </Box>
    </Box>
  )
}
```

**Comment by vadimdemedes** [comment]
I'm going to check it out, but I was able to get the correct behavior by removing the unnecessary nested `<Text>` component:

```jsx
<Box flexDirection="column" minHeight={6} width={30}>
  <Box width={30}>
    <Text>{letters.join('')}</Text>
  </Box>
</Box>
```

**Comment by vadimdemedes** [comment]
Fixed via https://github.com/vadimdemedes/ink/commit/5f080faed173b113b49f556f043027fec8c75224. See commit message for detailed explanation of the root issue. Published that fix in `3.0.0-5`. Thanks for reporting!
